src/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -892,5 +892,45 @@ public static void Uri_ExtremelyLongScheme_ThrowsUriFormatException()
             string largeString = new string('a', 1_000_000) + ":"; // 2MB is large enough to cause a stack overflow if we stackalloc the scheme buffer.
             Assert.Throws<UriFormatException>(() => new Uri(largeString));
         }
+
+        [Fact]
+        public static void Uri_HostTrailingSpaces_SpacesTrimmed()
+        {
+            string host = "www.contoso.com";
+            Uri u = new Uri($"http://{host}     ");
+
+            Assert.Equal($"http://{host}/", u.AbsoluteUri);
+            Assert.Equal(host, u.Host);
+        }
+
+        [Fact]
+        public static void Uri_PortTrailingSpaces_SpacesTrimmed()
+        {
+            int port = 1234;
+            Uri u = new Uri($"http://www.contoso.com:{port}     ");
+
+            Assert.Equal($"http://www.contoso.com:{port}/", u.AbsoluteUri);
+            Assert.Equal(port, u.Port);
+        }
+
+        [Fact]
+        public static void Uri_PathTrailingSpaces_SpacesTrimmed()
+        {
+            string path = "/path/";
+            Uri u = new Uri($"http://www.contoso.com{path}     ");
+
+            Assert.Equal($"http://www.contoso.com{path}", u.AbsoluteUri);
+            Assert.Equal(path, u.AbsolutePath);
+        }
+
+        [Fact]
+        public static void Uri_QueryTrailingSpaces_SpacesTrimmed()
+        {
+            string query = "?query";
+            Uri u = new Uri($"http://www.contoso.com/{query}     ");
+
+            Assert.Equal($"http://www.contoso.com/{query}", u.AbsoluteUri);
+            Assert.Equal(query, u.Query);
+        }
```

Comment:
Do we already have tests for:
- Spaces after the colon and before the port, e.g. `http://www.contoso.com:  80`?
- Spaces splitting the port, e.g. `http://www.contoso.com:8 0`?
- Non-space chars after the port, e.g. `http://www.contoso.com:80a`?
- No port but a port delimiter, e.g. `http://www.contoso.com:`?
- No port but a port delimiter followed by whitespace, e.g. `http://www.contoso.com:   `?

Reply:
Not as far as I can tell. I'll add these, and check the behavior before and after my fix.